src/plugins/mer/merbuildenginedetailswidget.cpp

```diff
@@ -402,5 +414,14 @@ void MerBuildEngineDetailsWidget::onWwwProxyExcludesEdited(const QString &s)
     emit wwwProxyChanged(m_wwwProxy, m_wwwProxyServerList, m_wwwProxyExcludes);
 }
 
+void MerBuildEngineDetailsWidget::onLocalhostAvailablePortsStateChanged()
+{
+    const MerLocalhostAvailablePorts &ports = MerLocalhostAvailablePorts::instance();
+    const bool available = (ports.state() == MerLocalhostAvailablePorts::Valid)
+            ? ports.isAvailable(m_ui->sshPortSpinBox->value())
+            : true;
+    m_ui->sshPortAvailableLabel->setHidden(available);
```

Comment:
Doesn't the label appear inadvertently whenever the VM is on? Shouldn't we have something like `setHidden(!vmOff || available)` here?

Reply:
I am using different label for this. Now there are two labels: sshPortInfoLabel and sshPortAvailableLabel. I am implemented sshPortInfoLabel label in order not to complicate the logic.

Reply:
yup, I know there are two labels. I wonder if you open the option page when the VM is running, and so the currently configured port is in use (by the VM itself), doesn't the `sshPortAvailableLabel` appear inadvertently?

Reply:
You are right. My bad. I'll fix it.

Reply:
Done.